packages/utils/src/collections.ts

```diff
@@ -39,3 +41,43 @@ export function intersection<T>(arrays: T[][]): T[] {
 
   return res
 }
+
+// List of sorted from greater to smaller
+
+export interface Item {
+  address: string
+  value: BigNumber
+}
+
+const nilAddress = '0x0000000000000000000000000000000000000000'
+
+function insert(sortedList: Item[], change: Item) {
+  const oldIdx = sortedList.findIndex((a) => a.address === change.address)
+  sortedList.splice(oldIdx, 1)
+  const newIdx = sortedList.findIndex((a) => a.value.lt(change.value))
+  if (newIdx === -1) {
+    sortedList.push(change)
+  } else {
+    sortedList.splice(newIdx, 0, change)
+  }
+}
+
+export function linkedListChanges(sortedList: Item[], changeList: Item[]) {
```

Comment:
This is super useful! 👏 

Love that it doesn't rely on sorting, since we've hit bugs from using unstable sorting algorithms in the past.

Can you create an issue to replace `findLesserAndGreater` in contractkit with this?